src/dbnode/storage/index/block.go

```diff
@@ -494,6 +496,7 @@ func (b *block) queryWithSpan(
 
 	// Add last batch to results if remaining.
 	if len(batch) > 0 {
+		docsCount += len(batch)
```

Comment:
Hm, I think `docsCount` would've been incremented since we increment each time we add to the batch above?  This is just calling `addQueryResults(...)` for the last batch that might not have been flushed.